src/backend/distributed/commands/dependencies.c

```diff
@@ -0,0 +1,164 @@
+/*-------------------------------------------------------------------------
+ *
+ * dependencies.c
+ *    Commands to create dependencies of an object on all workers.
+ *
+ * Copyright (c) 2019, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "catalog/dependency.h"
+#include "catalog/objectaddress.h"
+
+#include "distributed/dist_catalog/dependency.h"
+#include "distributed/connection_management.h"
+#include "distributed/worker_manager.h"
+#include "distributed/metadata_sync.h"
+#include "distributed/commands.h"
+#include "distributed/remote_commands.h"
+#include "distributed/dist_catalog/distobject.h"
+
+static List * GetDependencyCreateDDLCommands(const ObjectAddress *dependency);
+
+
+/*
+ * EnsureDependenciesExists finds all the dependencies that we can distribute and makes
+ * sure these are available on all workers. If not available they will be created on the
+ * workers via a separate session that will be committed directly so that the objects are
+ * visible to potentially multiple sessions creating the shards.
+ */
+void
+EnsureDependenciesExistsOnAllNodes(const ObjectAddress *target)
+{
+	const uint64 connectionFlag = FORCE_NEW_CONNECTION;
+	ListCell *dependencyCell = NULL;
+
+	List *dependencies = NIL;
+	List *connections = NULL;
+	ListCell *connectionCell = NULL;
+
+	/* collect all dependencies in creation order and get their ddl commands */
+	GetDependenciesForObject(target, &dependencies);
+
+	/* create all dependencies on all nodes and mark them as distributed */
+	foreach(dependencyCell, dependencies)
```

Comment:
function complexity

unnest loop
 - append all command to 1 list
    early escape if list is empty
 - connect to nodes
 - enter local entries
 - disconnect from nodes